Notes for 19/09/2025 Developer Call (#192)
gammapy/gammapy-meetings: dev-meetings/2025/2025-09-19/README.md

Meeting to get overview over the issues and coordinate how to solve them. One big challenge is to manage the diverse environments used in the various gammapy-recipes, perhaps pixi could be used to help this.

## Preparation of user call
AS shares slides. As most people do know gammapy already, no general introduction planned. Brief introduction of the new features (sensitivity calculation for analysis, full-vectorized error bar calculation, ...), Improved documentation, Outlook on plans for upcoming user calls on more specialised topics.

RT points out that as current release is new milestone, amount of work between LST 1.0 and 2.0 could be shown, incl. list of people that contributed. Also effort in meantime could be included, like the work on new datamodel.

A discussion about the possibility of asking for user input on the priorities for further features. However, not entirely simple to have right balance. 
RT summarises the message should be: that we are prioritising among too many features being requested, and if people want something on low priority they are encouraged to contribute code themselves. But users can contact us to receive support and guidance

> There will be a tutorial on bayesian sampling by Fabio Pintore and one on Fermi-analysis by QR.

Reviewer: ```suggestion
There will be a tutorial on bayesian sampling by Fabio Acero and one on Fermi-analysis by QR.
```

RT: TB, LS and me discussed about dev-call. We spend too much time on specific PR and could instead go in detail on specific PR, if needed. Important is that things are assigned, that reviews are being made and issues being tackled. Three main tasks: bug tracking, devops, documentation. For task, one person could give update. We could turn with sharing the dev-meeting in the future.

## Reports
RT has created several github projects to organise the bug tracking, the devops issues, and the documentation work.

### [Bug Tracker](https://github.com/orgs/gammapy/projects/36/views/1)

* [#6143](https://github.com/gammapy/gammapy/issues/6143) is assigned to AS
* [#6138](https://github.com/gammapy/gammapy/issues/6138) is assigned to RT
* [#6127](https://github.com/gammapy/gammapy/issues/6127) reclassified as a infra issue and moved to devops topic
* [#6118](https://github.com/gammapy/gammapy/issues/6118) reclassified as cleanup, connected to gammapy.maps, put into wishlist milestone
* [#6117](https://github.com/gammapy/gammapy/issues/6117) is assigned to KF, moved to documentation topic

### [DevOps](https://github.com/orgs/gammapy/projects/31)
#### Using sonarcube for quality assurance

> DM is working on adding a sonarcube workflow to the gammapy existing CI system. Hopefully done by next week.

Reviewer: ```suggestion
DM is working on adding a sonarcube workflow to the gammapy existing CI system [#6150](https://github.com/gammapy/gammapy/pull/6150). Hopefully done by next week.
```

> #### Discussion on how we want to proceed with towncrier (Draft PR #6130). 

Reviewer: ```suggestion
#### Discussion on how we want to proceed with towncrier (Draft PR [#6130](https://github.com/gammapy/gammapy/pull/6130)). 
```

By making developers create changelog files following a specific format, towncrier can automatically generate a well formatted changelog, including grouping several changes into sections. 

KF reports: Implementation will try to follow as closely as possible how the manually created changelogs currently look. For each PR, now rst-file has to be added. Naming-scheme: "Number of PR . Name. rst", e.g.: "1.infrastructure.rst".
KF thinks there needs to be some discussion on how we want to group changes, as this determines the naming scheme developers have to follow.
QR asks about case of multiple tags on an PR: either the developer decides which single tag to use when naming the rst-file, or you add several files, but it is not clear that actually works.

> RT summarises: We stick with 'improvement'-, 'bugfix'-, 'performance'- entries as currently, decided that should have changelog workflow 

Reviewer: ```suggestion
RT summarises: We stick with 'improvement'-, 'bugfix'-, 'performance'- entries as currently, we also should have a changelog workflow to make sure PRs requiring a changelog entry actually have one. One can use a dedicated label to mark them.
```

> Feature, bug, api change are all labels that will require a changelog, all others will need changelogs as needed.

Reviewer: ```suggestion
Conversely, considering that `feature`, `bug`, `api-change` are all labels that will require a changelog entry we could use the presence of one of these on a PR to trigger the workflow. PRs with different labels  others will need changelogs on a case by case basis.
```
